In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#Reading dataset into python environment

In [2]:
data=pd.read_excel("Dataset_w6.xlsx")
data

country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0              652864.0                         35530.0   
1               28748.0                          2930.0   
2             2381741.0                         41318.0   
3             1246700.0                         29784.0   
4                 442.0                           102.0   
..                  ...                             ...   
191            912050.0                         31977.0   
192            330967.0                         95541.0   
193            527968.0                         28250.0   
194            752612.0                         17094.0   
195            390757.0                         16530.0   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                           125.0   
1                          3984.2                           134.0   
2                          4154.1                           160.0   
3                          4714.1                           175.0   
4                         14764.5                            88.0   
..                            ...                             ...   
191                       11068.9                           118.0   
192                        2067.9                           136.0   
193                        1106.4                           136.0   
194                        1311.1                           179.0   
195                         890.4                            99.0   

     International trade: Imports (million US$)  
0                                        3568.0  
1                                        4669.0  
2                                       47091.0  
3                                        8790.0  
4                                         491.0  
..                                          ...  
191                                     16324.0  
192                                    174111.0  
193                                      6861.0  
194                                      7442.0  
195                                      5212.0  

[196 rows x 7 columns]

In [3]:
data.columns

Index(['country', 'Region', 'Surface area (km2)',
       'Population in thousands (2017)', 'GDP per capita (current US$)',
       'Agricultural production index ',
       'International trade: Imports (million US$)'],
      dtype='object')

In [4]:
data.describe()

Surface area (km2)  Population in thousands (2017)  \
count        1.960000e+02                    1.960000e+02   
mean         6.787071e+05                    3.796658e+04   
std          1.914200e+06                    1.434228e+05   
min          2.600000e+01                    5.000000e+00   
25%          2.084900e+04                    1.462000e+03   
50%          1.136275e+05                    7.677500e+03   
75%          5.077380e+05                    2.624075e+04   
max          1.709825e+07                    1.409517e+06   

       GDP per capita (current US$)  Agricultural production index   \
count                    196.000000                      196.000000   
mean                   13727.343367                      117.066327   
std                    18995.593914                       25.521247   
min                      144.500000                       34.000000   
25%                     1840.725000                      102.000000   
50%                     5463.450000                      114.000000   
75%                    15849.550000                      133.250000   
max                   100160.800000                      199.000000   

       International trade: Imports (million US$)  
count                                1.960000e+02  
mean                                 7.996138e+04  
std                                  2.340416e+05  
min                                  1.200000e+01  
25%                                  2.162500e+03  
50%                                  8.255500e+03  
75%                                  4.612275e+04  
max                                  2.249661e+06

In [5]:
data.head(5)

country          Region  Surface area (km2)  \
0          Afghanistan    SouthernAsia            652864.0   
1              Albania  SouthernEurope             28748.0   
2              Algeria  NorthernAfrica           2381741.0   
3               Angola    MiddleAfrica           1246700.0   
4  Antigua and Barbuda       Caribbean               442.0   

   Population in thousands (2017)  GDP per capita (current US$)  \
0                         35530.0                         623.2   
1                          2930.0                        3984.2   
2                         41318.0                        4154.1   
3                         29784.0                        4714.1   
4                           102.0                       14764.5   

   Agricultural production index   International trade: Imports (million US$)  
0                           125.0                                      3568.0  
1                           134.0                                      4669.0  
2                           160.0                                     47091.0  
3                           175.0                                      8790.0  
4                            88.0                                       491.0

1. Find out how many countries are there in the dataset.

In [6]:
data['country'].value_counts().sum()

196

2. Find out how many countries with respect to the region are available in the
dataset

In [7]:
data["Region"].value_counts()

WesternAsia          18
EasternAfrica        17
Caribbean            16
WesternAfrica        15
SouthAmerica         12
SouthernEurope       12
South-easternAsia    11
EasternEurope        10
NorthernEurope       10
SouthernAsia          9
MiddleAfrica          9
CentralAmerica        8
WesternEurope         7
EasternAsia           7
SouthernAfrica        5
Polynesia             5
NorthernAfrica        5
Melanesia             5
CentralAsia           5
NorthernAmerica       4
Micronesia            4
Oceania               2
Name: Region, dtype: int64

In [8]:
len(data["Region"].value_counts())

22

196 countries with respect to 22 regions are there in the dataset.

3.On average which region has the highest


a. GDP per capita (current US$)

In [12]:
data1=data[['Region','GDP per capita (current US$)']].groupby('Region').mean()
data1=data1.sort_values('GDP per capita (current US$)',ascending=True)
data1

GDP per capita (current US$)
Region                                         
WesternAfrica                       1013.140000
EasternAfrica                       2088.852941
SouthernAsia                        2968.400000
MiddleAfrica                        3928.833333
NorthernAfrica                      3934.960000
SouthernAfrica                      4191.900000
CentralAsia                         4329.880000
Micronesia                          5008.300000
CentralAmerica                      6348.137500
SouthAmerica                        8499.775000
Polynesia                           8686.520000
Melanesia                           9261.960000
EasternEurope                       9292.490000
South-easternAsia                  10299.027273
SouthernEurope                     14462.741667
Caribbean                          15135.562500
WesternAsia                        17013.938889
EasternAsia                        27967.614286
NorthernEurope                     42098.220000
Oceania                            44823.250000
WesternEurope                      55387.128571
NorthernAmerica                    57659.000000

In [13]:
data1.nlargest(1,['GDP per capita (current US$)'])

GDP per capita (current US$)
Region                                       
NorthernAmerica                       57659.0

b. International trade: Imports (million US$)

In [14]:
data2=data[['Region','International trade: Imports (million US$)']].groupby('Region').mean()
data2=data2.sort_values('International trade: Imports (million US$)',ascending=False)
data2

International trade: Imports (million US$)
Region                                                       
NorthernAmerica                                 663552.250000
EasternAsia                                     451995.000000
WesternEurope                                   407498.428571
NorthernEurope                                  113407.200000
Oceania                                         112914.500000
South-easternAsia                                98324.636364
EasternEurope                                    85243.900000
SouthernEurope                                   76666.666667
SouthernAsia                                     58199.333333
CentralAmerica                                   57031.500000
WesternAsia                                      46025.555556
NorthernAfrica                                   35034.400000
SouthAmerica                                     32885.666667
SouthernAfrica                                   18005.600000
CentralAsia                                      12063.400000
WesternAfrica                                     4898.333333
EasternAfrica                                     4726.176471
MiddleAfrica                                      4381.222222
Caribbean                                         3057.312500
Melanesia                                         2896.400000
Polynesia                                          432.000000
Micronesia                                          99.000000

In [15]:
data2.nlargest(1,['International trade: Imports (million US$)'])

International trade: Imports (million US$)
Region                                                     
NorthernAmerica                                   663552.25

4. Find out which region has more consistent with respect to GDP per capita
(current US$)

In [16]:
data1=data[['Region','GDP per capita (current US$)']].groupby('Region').std()
data1=data1.sort_values('GDP per capita (current US$)',ascending=True)
data1

GDP per capita (current US$)
Region                                         
WesternAfrica                        815.095399
NorthernAfrica                       974.706575
SouthernAfrica                      2177.551120
SouthernAsia                        2855.708179
EasternAfrica                       3716.131770
CentralAsia                         4152.215490
CentralAmerica                      4180.440883
SouthAmerica                        4273.707394
Micronesia                          4820.669694
MiddleAfrica                        5274.846281
EasternEurope                       5415.488739
Polynesia                           6956.250286
Oceania                             9233.329638
SouthernEurope                      9541.703216
Melanesia                          13855.842686
Caribbean                          14579.371326
South-easternAsia                  16311.810656
WesternAsia                        18812.151459
NorthernEurope                     20671.030185
WesternEurope                      24770.467728
NorthernAmerica                    25749.619445
EasternAsia                        27485.120923

In [17]:
data1['GDP per capita (current US$)'].idxmin()

'WesternAfrica'